tools/notifyServicelogEvent.c

```diff
@@ -140,77 +143,56 @@ main(int argc, char *argv[])
         crm_err("Error: event_id is 0!");
-
         rc = 1;
         goto done;
     }
 
     rc = servicelog_open(&slog, 0);     /* flags is one of SL_FLAG_xxx */
-
-    if (!slog) {
-        crm_err("Error: servicelog_open failed, rc = %d", rc);
-
+    if (rc != 0) {
+        crm_err("Error: Failed to open the servicelog, rc = %d", rc);
         rc = 1;
         goto done;
     }
 
-    if (slog) {
-        rc = servicelog_event_get(slog, event_id, &event);
+    rc = servicelog_event_get(slog, event_id, &event);
+    if (rc != 0) {
+        crm_err("Error: Failed to get event from the servicelog, rc = %d", rc);
+        rc = 1;
+        goto done;
     }
 
-    if (rc == 0) {
-        STATUS status = STATUS_GREEN;
-        const char *health_component = "#health-ipmi";
-        const char *health_status = NULL;
-
-        crm_debug("Event id = %" U64T ", Log timestamp = %s, Event timestamp = %s",
-                  event_id, ctime(&(event->time_logged)), ctime(&(event->time_event)));
-
-        status = event2status(event);
-
-        health_status = status2char(status);
-
-        if (health_status) {
-            int attrd_rc;
-
-            // @TODO pass pcmk__node_attr_remote when appropriate
-            attrd_rc = pcmk__attrd_api_update(NULL, NULL, health_component,
-                                              health_status, NULL, NULL, NULL,
-                                              pcmk__node_attr_pattern);
-            if (attrd_rc == pcmk_rc_ok) {
-                crm_debug("Updating attribute %s=%s: %d",
-                          health_component, health_status, attrd_rc);
-            } else {
-                crm_err("Could not update %s=%s: %s (%d)",
-                        health_component, health_status, pcmk_rc_str(attrd_rc),
-                        attrd_rc);
-            }
-
-        } else {
-            crm_err("Error: status2char failed, status = %d", status);
-            rc = 1;
-        }
+    crm_debug("Event id = %" U64T ", Log timestamp = %s, Event timestamp = %s",
+              event_id, ctime(&(event->time_logged)),
+              ctime(&(event->time_event)));
+
+    health_status = status2char(event2status(event));
+
+    // @TODO pass pcmk__node_attr_remote when appropriate
+    rc = pcmk__attrd_api_update(NULL, NULL, health_component, health_status,
+                                NULL, NULL, NULL, pcmk__node_attr_pattern);
+    if (rc == pcmk_rc_ok) {
+        crm_debug("Updating attribute %s=%s: %d",
+                  health_component, health_status, rc);
     } else {
-        crm_err("Error: servicelog_event_get failed, rc = %d", rc);
+        crm_err("Could not update %s=%s: %s (%d)",
+                health_component, health_status, pcmk_rc_str(rc), rc);
     }
 
   done:
-    if (event) {
+    if (event != NULL) {
         servicelog_event_free(event);
     }
 
-    if (slog) {
+    if (slog != NULL) {
         servicelog_close(slog);
     }
 
     closelog();
-
     return rc;
```

Comment:
I'll update this to use an exit code momentarily. It is done in the GOption commit currently

Reply:
Simpler to just add `rc = 1` in the last `else` block; pushed that just now.